File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l=0;
        int r=l+1;
        int n=prices.size();
        int p=0;
        int mp=0;
        while(r<n){
        
            if(prices[l]<prices[r]){
                p=prices[r]-prices[l];
                mp=max(p,mp);
            }
            else {l=r;}
            r++;
        }
        return mp;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        mini=prices[0]
        maxprofit=0
        for i in range(1,n):
            mini=min(mini,prices[i])
            profit=prices[i]-mini
            maxprofit=max(maxprofit,profit)
        return maxprofit